src/basic/socket-util.c

```diff
@@ -1319,6 +1320,52 @@ int flush_accept(int fd) {
         }
 }
 
+int flush_mqueue(int fd) {
+        _cleanup_free_ char *buf = NULL;
+        struct mq_attr attr;
+        unsigned pri;
+        int r, count = 0;
+
+        assert(fd >= 0);
+
+        /* Similar to flush_fd() but flushes all messages from a POSIX message queue. */
+
+        if (mq_getattr(fd, &attr) < 0)
+                return -errno;
+
+        /* Buffer must be at least as large as mq_msgsize. */
+        buf = malloc(attr.mq_msgsize);
+        if (!buf)
+                return -ENOMEM;
+
+        for (;;) {
+                ssize_t l;
+
+                r = fd_wait_for_event(fd, POLLIN, 0);
```

Comment:
can you prefix the last argument with `/* timeout= */` please. we do that in new code whenever a func parameter is otherwise too opaque

Reply:
Good catch, done.